src/lib/SapphireClient.ts

```diff
@@ -107,8 +129,39 @@ export class SapphireClient extends Client {
 	 * @retrun Token of the account used.
 	 */
 	public async login(token?: string) {
+		for (const plugin of SapphireClient.plugins(PluginHook.PreLogin)) {
+			plugin.hook.call(this);
+			this.emit(Events.PluginLoaded, plugin.type, plugin.name);
+		}
+
 		await Promise.all([...this.stores].map((store) => store.loadAll()));
-		return super.login(token);
+		const login = await super.login(token);
+
+		for (const plugin of SapphireClient.plugins(PluginHook.PostLogin)) {
+			plugin.hook.call(this);
+			this.emit(Events.PluginLoaded, plugin.type, plugin.name);
+		}
+
+		return login;
+	}
+
+	protected static _plugins: Set<SapphirePluginHookEntry> = new Set();
```

Comment:
```suggestion
	protected static _plugins = new Set<SapphirePluginHookEntry>();
```

Reply:
Also, may the field be called `registeredPlugins` or something? It's protected, not private.